Assets/MixedRealityToolkit.SDK/Features/Utilities/Solvers/SolverHandler.cs

```diff
@@ -134,17 +176,62 @@ public bool UpdateSolvers
         /// </summary>
         public float DeltaTime { get; set; }
 
-        private bool RequiresOffset => AdditionalOffset.sqrMagnitude != 0 || AdditionalRotation.sqrMagnitude != 0;
+        /// <summary>
+        /// The target transform that the solvers will act upon.
+        /// </summary>
+        public Transform TransformTarget
+        {
+            get
+            {
+                if (IsInvalidTracking())
+                {
+                    RefreshTrackedObject();
+                }
 
-        protected readonly List<Solver> solvers = new List<Solver>();
+                return trackingTarget?.transform;
+            }
+        }
 
-        private float lastUpdateTime;
 
-        private GameObject transformWithOffset;
+        // Stores currently attached hand if valid (only possible values Left, Right, or None)
+        protected Handedness currentTrackedHandedness = Handedness.None;
+
+        /// <summary>
+        /// Currently tracked hand or motion controller if applicable
+        /// </summary>
+        public Handedness CurrentTrackedHandedness
+        {
+            get
+            {
+                return currentTrackedHandedness;
+            }
+        }
+
+        // Hidden GameObject managed by this component and attached as a child to the tracked target type (i.e head, hand etc)
+        private GameObject trackingTarget;
+
+        private float lastUpdateTime;
 
         private IMixedRealityHandJointService HandJointService => handJointService ?? (handJointService = (InputSystem as IMixedRealityDataProviderAccess)?.GetDataProvider<IMixedRealityHandJointService>());
         private IMixedRealityHandJointService handJointService = null;
 
+        private IMixedRealityInputSystem inputSystem = null;
+
+        /// <summary>
+        /// The active instance of the input system.
+        /// </summary>
+        protected IMixedRealityInputSystem InputSystem
+        {
+            get
+            {
+                if (inputSystem == null)
+                {
+                    MixedRealityServiceRegistry.TryGetService<IMixedRealityInputSystem>(out inputSystem);
+                }
+                return inputSystem;
+            }
+        }
```

Comment:
I wish your static helper thingy went in....

Reply:
Me too, I am still working on it. Just trying to resolve some gaps in MRTK.cs and the Registry